File: web/src/api/frontend/compliance/diagnosis.ts
/**
 * 前台合规诊断公开 API（无需登录）
 * @module api/frontend/compliance/diagnosis
 */
import { siteRequest } from '@/utils/http'

/** 月收入区间 */
export type MonthlyIncomeRange = '0-2万' | '2-5万' | '5-15万' | '15万+'

/** 现有主体类型 */
export type ExistingEntity = 'none' | 'individual' | 'company' | 'other'

/** 报税状态 */
export type HasFiledTax = 'yes' | 'no' | 'unsure'

/** 推荐方案 */
export type RecommendedPlan = 'opc' | 'individual' | 'labor' | 'transitional' | 'none'

/** 单方案税负对比项 */
export interface TaxComparisonItem {
  /** 方案标识：none / labor / individual / opc */
  plan: string
  /** 方案名称 */
  label: string
  /** 预估年税负（元） */
  annualTax: number
  /** 综合税负率 */
  taxRate: number
  /** 是否为推荐方案 */
  recommended?: boolean
  /** 是否为高危警告（不报税列） */
  warning?: boolean
  /** 补充说明 */
  note?: string
}

/** 税负对比结果 */
export interface TaxComparison {
  annualIncome: number
  annualCost: number
  items: TaxComparisonItem[]
}

/** 诊断问卷提交参数 */
export interface DiagnosisSubmitParams {
  platforms: string[]
  monthlyIncomeRange: MonthlyIncomeRange
  annualCostEstimate?: number
  existingEntity: ExistingEntity
  hasFiledTax: HasFiledTax
  taxBureauContact: boolean
  notes?: string
  costBreakdown?: Record<string, number>
}

/** 诊断提交响应 */
export interface DiagnosisSubmitResult {
  id: number | string
  recommendedPlan: RecommendedPlan
  taxComparison: TaxComparison
  reasons?: string[]
  assumptionHints?: string[]
}

/** 税负计算器参数 */
export interface TaxCalculatorParams {
  annualIncome: number
  annualCost: number
  diagnosisId?: number | string
}

/** 税负计算器响应 */
export interface TaxCalculatorResult {
  taxComparison: TaxComparison
  recommendedPlan?: RecommendedPlan
  reasons?: string[]
  assumptionHints?: string[]
}

/** 服务套餐 */
export interface ServicePlan {
  id: number | string
  name: string
  tier: 'basic' | 'advanced' | 'premium'
  monthlyPrice: number | null
  priceLabel?: string
  features: string[]
  recommended?: boolean
}

/** 套餐列表响应 */
export interface ServicePlansResult {
  list: ServicePlan[]
}

const DIAGNOSIS_RESULT_KEY = 'compliance_diagnosis_result'

/** 缓存诊断结果（供结果页刷新后读取） */
export function cacheDiagnosisResult(id: number | string, result: DiagnosisSubmitResult) {
  sessionStorage.setItem(`${DIAGNOSIS_RESULT_KEY}_${id}`, JSON.stringify(result))
}

/** 读取缓存的诊断结果 */
export function getCachedDiagnosisResult(id: number | string): DiagnosisSubmitResult | null {
  const raw = sessionStorage.getItem(`${DIAGNOSIS_RESULT_KEY}_${id}`)
  if (!raw) return null
  try {
    return JSON.parse(raw) as DiagnosisSubmitResult
  } catch {
    return null
  }
}

/** 提交合规诊断问卷 */
export function submitDiagnosis(params: DiagnosisSubmitParams) {
  return siteRequest.post<DiagnosisSubmitResult>({
    url: '/compliance/diagnosis',
    data: params
  })
}

/** 税负计算器（可调年收入/成本） */
export function calculateTax(params: TaxCalculatorParams) {
  return siteRequest.post<TaxCalculatorResult>({
    url: '/compliance/calculator',
    data: params
  })
}

/** 获取服务套餐列表 */
export async function fetchServicePlans(): Promise<ServicePlansResult> {
  const res = await siteRequest.get<ServicePlansResult | ServicePlan[]>({
    url: '/compliance/plans'
  })
  if (Array.isArray(res)) return { list: res }
  return { list: (res as ServicePlansResult)?.list ?? [] }
}

File: server/internal/model/input/compliancein/diagnosis.go
package compliancein

// TaxComparisonItem 单方案税负对比项
type TaxComparisonItem struct {
	Plan        string  `json:"plan"`
	Label       string  `json:"label"`
	AnnualTax   float64 `json:"annualTax"`
	TaxRate     float64 `json:"taxRate"`
	Recommended bool    `json:"recommended,omitempty"`
	Warning     bool    `json:"warning,omitempty"`
	Note        string  `json:"note,omitempty"`
}

// TaxComparison 税负对比结果
type TaxComparison struct {
	AnnualIncome float64             `json:"annualIncome"`
	AnnualCost   float64             `json:"annualCost"`
	Items        []TaxComparisonItem `json:"items"`
}

// DiagnosisSubmitInp 诊断问卷提交入参
type DiagnosisSubmitInp struct {
	MemberId            uint64
	Platforms           []string
	MonthlyIncomeRange  string
	AnnualCostEstimate  float64
	ExistingEntity      string
	HasFiledTax         string
	TaxBureauContact    bool
	Notes               string
	CostBreakdown       map[string]float64
}

// DiagnosisSubmitModel 诊断提交出参
type DiagnosisSubmitModel struct {
	Id              uint64        `json:"id"`
	RecommendedPlan string        `json:"recommendedPlan"`
	TaxComparison   TaxComparison `json:"taxComparison"`
	Reasons         []string      `json:"reasons,omitempty"`
	AssumptionHints []string      `json:"assumptionHints,omitempty"`
}

// TaxCalculatorInp 税负计算器入参
type TaxCalculatorInp struct {
	AnnualIncome float64
	AnnualCost   float64
	DiagnosisId  uint64
	TaxBureauContact bool
}

// TaxCalculatorModel 税负计算器出参
type TaxCalculatorModel struct {
	TaxComparison   TaxComparison `json:"taxComparison"`
	RecommendedPlan string        `json:"recommendedPlan,omitempty"`
	Reasons         []string      `json:"reasons,omitempty"`
	AssumptionHints []string      `json:"assumptionHints,omitempty"`
}

// ServicePlanItem 服务套餐项
type ServicePlanItem struct {
	Id            uint64   `json:"id"`
	Name          string   `json:"name"`
	Tier          string   `json:"tier"`
	MonthlyPrice  *float64 `json:"monthlyPrice"`
	PriceLabel    string   `json:"priceLabel,omitempty"`
	Features      []string `json:"features"`
	Recommended   bool     `json:"recommended,omitempty"`
}

// ServicePlansModel 套餐列表出参
type ServicePlansModel struct {
	List []ServicePlanItem `json:"list"`
}

// DiagnosisHistoryInp 诊断历史查询入参
type DiagnosisHistoryInp struct {
	MemberId uint64
	Page     int
	PageSize int
}

// DiagnosisHistoryItem 诊断历史项
type DiagnosisHistoryItem struct {
	Id                 uint64        `json:"id"`
	Platforms          []string      `json:"platforms"`
	MonthlyIncomeRange string        `json:"monthlyIncomeRange"`
	AnnualCostEstimate float64       `json:"annualCostEstimate"`
	ExistingEntity     string        `json:"existingEntity"`
	HasFiledTax        string        `json:"hasFiledTax"`
	TaxBureauContact   bool          `json:"taxBureauContact"`
	RecommendedPlan    string        `json:"recommendedPlan"`
	TaxComparison      TaxComparison `json:"taxComparison"`
	CreatedAt          string        `json:"createdAt"`
}

// DiagnosisHistoryModel 诊断历史出参
type DiagnosisHistoryModel struct {
	List     []DiagnosisHistoryItem `json:"list"`
	Page     int                    `json:"page"`
	PageSize int                    `json:"pageSize"`
	Total    int                    `json:"total"`
}
